ocaml/xapi/xapi_clustering.ml

```diff
@@ -17,6 +17,13 @@ open Cluster_interface
 module D=Debug.Make(struct let name="xapi_clustering" end)
 open D
 
+(* Called by Cluster.create/destroy and Cluster_host.enable/disable *)
+let set_ha_cluster_stack_if_ha_enabled ~__context =
+  let pool = Helpers.get_pool ~__context in
+  let value = Cluster_stack_constraints.choose_cluster_stack ~__context in
+  if Db.Pool.get_ha_enabled ~__context ~self:pool then
```

Comment:
I think you should drop the `if` we should always set it, because you can query this parameter with `xe`.
If you enable corosync and it shows ha cluster stack as `xhad`, then one might think that enabling HA could use `xhad` if a suitable heartbeat SR is chosen, but that is wrong, therefore once corosync is enabled we should update the acceptable choice for ha cluster stack.